packages/perseus/src/widgets/radio/radio-component.new.tsx

```diff
@@ -56,85 +52,58 @@ export type RadioChoiceWithMetadata = PerseusRadioChoice & {
+    showSolutions: "none",
+};
+
 /**
  * This component is a duplicate of the Radio component in radio.tsx
  * for the Radio Revitalization Project. (LEMS-2933)
- * This component will be split up to move the rendering logic to a new functional component.
- * This file will be renamed to radio.class.tsx and will supercede radio-component.tsx when
- * the feature flag is no longer needed.
+ * This component has been split up to move the rendering logic to a new functional component.
+ * The class methods are now in radio.class.new.tsx.
  *
  * TODO(LEMS-2994): Clean up this file.
  */
-class Radio extends React.Component<Props> implements Widget {
-    static contextType = PerseusI18nContext;
-    declare context: React.ContextType<typeof PerseusI18nContext>;
-
-    // @ts-expect-error - TS2564 - Property 'focusFunction' has no initializer and is not definitely assigned in the constructor.
-    focusFunction: FocusFunction;
-
-    static defaultProps: DefaultProps = {
-        choices: [],
-        multipleSelect: false,
-        countChoices: false,
-        deselectEnabled: false,
-        linterContext: linterContextDefault,
-        showSolutions: "none",
-    };
-
-    static getUserInputFromProps(
-        props: Props,
-        unshuffle: boolean = true,
-    ): PerseusRadioUserInput {
-        // Return checked inputs in the form {choicesSelected: [bool]}. (Dear
-        // future timeline implementers: this used to be {value: i} before
-        // multiple select was added)
-        if (props.choiceStates) {
-            const choiceStates = props.choiceStates;
-            const choicesSelected = choiceStates.map(() => false);
-
-            for (let i = 0; i < choicesSelected.length; i++) {
-                const index = unshuffle ? props.choices[i].originalIndex : i;
-
-                choicesSelected[index] = choiceStates[i].selected;
-            }
-
-            return {
-                choicesSelected,
-            };
-            // Support legacy choiceState implementation
-        }
-        /* c8 ignore if - props.values is deprecated */
-        const {values} = props;
-        if (values) {
-            const choicesSelected = [...values];
-            const valuesLength = values.length;
-
-            for (let i = 0; i < valuesLength; i++) {
-                const index = unshuffle ? props.choices[i].originalIndex : i;
-                choicesSelected[index] = values[i];
-            }
-            return {
-                choicesSelected,
-            };
-        }
-        // Nothing checked
-        return {
-            choicesSelected: props.choices.map(() => false),
-        };
-    }
-
-    componentDidUpdate(prevProps: Props) {
-        if (
-            this.props.showSolutions === "selected" &&
-            prevProps.showSolutions !== "selected"
-        ) {
-            this.showRationalesForCurrentlySelectedChoices(this.props);
-        }
-    }
+const RadioComponent = (props: Props) => {
```

Comment:
Not sure what's the team's convention, but it would be cleaner to have it like this instead of adding the const in line 74

```suggestion
const RadioComponent = ({
    choices,
    multipleSelect = defaultProps.multipleSelect,
    countChoices = defaultProps.countChoices,
    deselectEnabled = defaultProps.deselectEnabled,
    linterContext = defaultProps.linterContext,
    showSolutions = defaultProps.showSolutions,
    static: isStatic,
    reviewModeRubric,
    reviewMode,
    apiOptions,
    isLastUsedWidget,
    questionCompleted,
    findWidgets,
    onChange,
    trackInteraction,
    numCorrect,
}: RadioComponentProps) => {
```

Reply:
I wasn't thinking about the type updates in this PR but I can absolutely swap those over! 

However I've left the `props: RadioComponentProps` the way it is (albeit with your slightly clearer name), as I believe that's the standard we're using in all the other widgets. I'm happy to discuss during standup today though, as I don't have a hard line on this.

Reply:
@SonicScrewdriver let's disregard this comment and follow the team's standard and all other widget usages.